lib/aggregate/aggregate_store.rb

```diff
@@ -81,6 +81,15 @@ def self.included(model_class)
       model_class.extend ClassMethods
     end
 
+    ActiveRecordHelpers::Version.if_version(
+      active_record_gt_4: -> do
+        def changed_for_autosave?
+          defined?(super) or raise NoMethodError, "undefined method 'changed_for_autosave?' for #{self}"
```

Comment:
I don't quite understand why were are raising? This behavior is not present in #changed? rails 4, it simply short cuircuts to @ changed

Reply:
IMO `changed?` sort of has two responsibilities based on its current implementation:
1. It's overriding the `changed?` method for active record objects
2. It's offering a `changed?` method for non-active record objects, and given that the method name is generic I'm assuming it's helpful to have this as a method.

`changed_for_autosave?` is more similar to `saved_changes?` defined below (`saved_changes?` also raises if there's no `super` defined), where it wouldn't make sense to have an extra purpose like option 2 for non-active record objects, so if a non-active record object tries to call it we should know immediately with the exception.